### PR TITLE
Implement readlink

### LINKED ISSUE #111: Implement readlink
I'll work on this.

### PR DESCRIPTION
**Pull request by lht**
Fixes #111

### CODE REVIEW

File: src/readlink/readlink.rs [hunk 1]
```diff
@@ -0,0 +1,209 @@
+#![crate_name = "readlink"]
+
+/*
+ * This file is part of the uutils coreutils package.
+ *
+ * For the full copyright and license information, please view the LICENSE
```
Comment (Arcterus) [inline src/readlink/readlink.rs]: Would you mind adding your name in here?

File: src/readlink/readlink.rs [hunk 2]
```diff
@@ -0,0 +1,209 @@
...
+use std::vec::Vec;
+
+use CanonicalizeMode::{None, Normal, Existing, Missing};
+
+
+#[path = "../common/util.rs"]
+mod util;
+
+static NAME: &'static str = "readlink";
+static VERSION:  &'static str = "0.0.1";
```
Comment (Arcterus) [inline src/readlink/readlink.rs]: These should probably be `const` rather than `static`.

File: src/readlink/readlink.rs [hunk 3]
```diff
@@ -0,0 +1,209 @@
...
+enum CanonicalizeMode {
+    None,
+    Normal,
+    Existing,
+    Missing,
+}
+
+
+fn resolve(original: &Path) -> io::IoResult<Path> {
+    static MAX_LINKS_FOLLOWED: uint = 256;
```
Comment (Arcterus) [inline src/readlink/readlink.rs]: This should also probably be `const`.

**Comment by Arcterus** [comment]
Sorry it took so long to get to this!  I was on my holiday break.

I think everything else is fine, but you should probably make sure the code still works with the latest Rust.

**Comment by lht** [comment]
Thanks for reviewing, Arcterus! And all comments are addressed.

**Comment by Arcterus** [comment]
Thanks!

**Pull request merged**